readthedocs/config/config.py

```diff
@@ -135,9 +139,43 @@ def catch_validation_error(self, key):
                 code=error.code,
                 error_message=str(error),
                 source_file=self.source_file,
-                source_position=self.source_position)
+                source_position=self.source_position
+            )
+
+    def validate(self):
+        raise NotImplementedError()
+
+    def __getattr__(self, name):
+        """Raise an error for unknown attributes."""
+        raise ConfigNotSupportedError(name)
+
+
+class BuildConfig(BuildConfigBase):
```

Comment:
Maybe `BuildConfigVersion1` or something to distinguish this from the v2 class name?

Reply:
Yeah, I want to change that when implementing the v2 and the _class selector_ depending on the version. That way the changes here aren't too big.